### PR TITLE
GODRIVER-1788 correctly replace errors in IndexView CreateMany

### PR DESCRIPTION
**Pull request by iwysiu**
After reading through, I'm pretty sure that this is the only spot that isn't replacing errors correctly. Also fixed the CreateMany commitQuorum tests while i was there.

### CODE REVIEW

File: mongo/integration/index_view_test.go [hunk 1]
```diff
@@ -301,13 +321,13 @@ func TestIndexView(t *testing.T) {
 				mtOpts := mtest.NewOptions().MinServerVersion(tc.minServerVersion).MaxServerVersion(tc.maxServerVersion)
 				mt.RunOpts(tc.name, mtOpts, func(mt *mtest.T) {
 					mt.ClearEvents()
-					_, err := mt.Coll.Indexes().CreateOne(mtest.Background, indexModel, tc.opts)
+					_, err := mt.Coll.Indexes().CreateMany(mtest.Background, []mongo.IndexModel{indexModel1, indexModel2}, tc.opts)
```
Comment (kevinAlbs) [inline mongo/integration/index_view_test.go:324]: Nice attention to detail! I suppose this was copy-pasta of the "create one" tests.

My preference is to put this change on a separate commit (though can be part of this PR) to keep the scope of the commit just to the bug it fixes. `git add -p` can be helpful to add just the part of the diff you want to a commit.

**Review by kevinAlbs — APPROVED** [review]
LGTM!

File: mongo/integration/index_view_test.go (hunk 1 above)
Comment (divjotarora) [reply mongo/integration/index_view_test.go:324]: I don't have a strong preference here, but does putting it in a separate commit matter if we're going to squash the commits when merging the PR anyway?

**Review by divjotarora — APPROVED** [review]

File: mongo/integration/index_view_test.go (hunk 1 above)
Comment (kevinAlbs) [reply mongo/integration/index_view_test.go:325]: No, if the commits are squashed when merging then there's no point to split it up. Maybe the thing to do is have separate PRs then. Multi-commit PRs often require force pushing to get the commits right, which isn't great.

In general, I think it's better to limit the scope of a commit for a bug fix, to make backporting easier. Drive by fixes could be done separate PRs.

At any rate, I'm fine with having this in the commit. This is not a large addition. So I'll resolve.
